zio-http/src/main/scala/zhttp/http/HttpContent.scala

```diff
@@ -10,4 +11,20 @@ sealed trait HttpContent[-R, +A] extends Product with Serializable
 object HttpContent {
   final case class Complete[A](data: A)                        extends HttpContent[Any, A]
   final case class Chunked[R, A](data: ZStream[R, Nothing, A]) extends HttpContent[R, A]
+
+  sealed trait FormDataContent extends Product with Serializable
+
+  // TODO: Make the environment R instead of hardcoded Blocking.
+  // It's set to Blocking for now because we need it when reading from a tmp file.
+  final case class FileData(name: String, contentType: String, content: ZStream[Blocking, Throwable, Byte])
+      extends FormDataContent
+  final case class AttributeData(name: String, content: ZStream[Blocking, Throwable, Byte]) extends FormDataContent
+
+  object MultipartFormData {
+    def empty: MultipartFormData = MultipartFormData(Map.empty, Map.empty)
+  }
+  final case class MultipartFormData(
+    attributes: Map[String, AttributeData],
+    files: Map[String, FileData],
```

Comment:
since the name is there in both `FileData` and `AttributeData` do we need a `Map` here?

Reply:
Thought the `Map` would be convenient for by-name lookup. Can make it a `Seq` as well but then you have to do a `find`. I personally like keeping the `name` in the `Data` types as well such that you can pass these around and keep everything together.